app/routes/datamart/land.py

```diff
@@ -41,28 +45,107 @@
 
 router = APIRouter()
 
+def _parse_dataset_versions(request: Request) -> Dict[str, str]:
```

Comment:
just curious, why'd you move these functions to the top? is it more standard to have private function first?

Reply:
Good question; it's because of how Python evaluates default values for formal parameters in function definitions.

Python evaluates a default value only once when the module is loaded. In this concrete example, `Depends` is called when the default is evaluated. At that point, `_parse_area_of_interest` and `_parse_dataset_versions` need to be defined. In the previous version, they were lower down in the module and not defined yet.